### PR TITLE
Add pagination to cleezy page

### PR DESCRIPTION
**Pull request by gursheyss**
https://streamable.com/e/8c2v7q

### CODE REVIEW

File: src/Pages/URLShortener/URLShortener.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (evanugarte) [inline src/Pages/URLShortener/URLShortener.js:1]: remove this + console log?

File: api/peripheral_api/routes/Cleezy.js [hunk 2]
```diff
@@ -33,13 +33,16 @@ router.get('/listAll', async (req, res) => {
     return res.sendStatus(UNAUTHORIZED);
   }
   try {
-    const response = await axios.get(CLEEZY_URL + '/list');
+    const page = req.query.page || 0;
```
Comment (evanugarte) [inline api/peripheral_api/routes/Cleezy.js:36]: if you wanna be cool
```suggestion
    const { page = 0 } = req.query;
```

File: src/Pages/URLShortener/URLShortener.js (hunk 1 above)
Comment (gursheyss) [reply src/Pages/URLShortener/URLShortener.js]: done

File: api/peripheral_api/routes/Cleezy.js (hunk 2 above)
Comment (gursheyss) [reply api/peripheral_api/routes/Cleezy.js]: done

File: api/peripheral_api/routes/Cleezy.js [hunk 3]
```diff
@@ -33,13 +33,16 @@ router.get('/listAll', async (req, res) => {
...
   }
   try {
-    const response = await axios.get(CLEEZY_URL + '/list');
+    const { page = 0 } = req.query;
+    const response = await axios.get(CLEEZY_URL + '/list', {
+      params: { page }
+    });
     const data = response.data;
-    const returnData = data.map(element => {
+    const returnData = data.data.map(element => {
```
Comment (evanugarte) [inline api/peripheral_api/routes/Cleezy.js:41]: we can do 
```js
const { data = [], total } = response;
res.json({
  data: data.map(...),
  total,
})
```

File: src/Pages/URLShortener/URLShortener.js [hunk 4]
```diff
@@ -102,6 +104,28 @@ export default function URLShortenerPage(props) {
     setAliasTaken(false);
   }, [alias]);
 
+  useEffect(() => {
+    getCleezyUrls(page);
+  }, [page]);
+
+  useEffect(() => {
+
+    const amountOfUsersOnCurrentPage = Math.min((page + 1) * rowsPerPage, allUrls.length);
```
Comment (evanugarte) [inline src/Pages/URLShortener/URLShortener.js:113]: nice copypasta
```suggestion
    const amountOfRowsOnCurrentPage = Math.min((page + 1) * rowsPerPage, allUrls.length);
```

File: src/Pages/URLShortener/URLShortener.js [hunk 5]
```diff
@@ -312,6 +370,7 @@ export default function URLShortenerPage(props) {
                 })}
```
Comment (joshua-demo) [inline src/Pages/URLShortener/URLShortener.js:370]: Can we add some css for the different columns to give them a fixed width? (lines 330-333 with the example on how to do it on line 332)

File: src/Pages/URLShortener/URLShortener.js [hunk 6]
```diff
@@ -121,6 +145,40 @@ export default function URLShortenerPage(props) {
     }
   }
 
+  function maybeRenderPagination() {
+    const amountOfUrlsOnPage = Math.min((page + 1) * rowsPerPage, allUrls.length);
+    const pageOffset = page * rowsPerPage;
+    const endingElementNumber = amountOfUrlsOnPage + pageOffset;
+    if (allUrls.length) {
```
Comment (evanugarte) [inline src/Pages/URLShortener/URLShortener.js:152]: how about 
```js
if (!allUrls.length) {
  return <></>
}
return (
  // the code to show the buttons
);
```

**Review by evanugarte — APPROVED** [review]

File: src/Pages/URLShortener/URLShortener.js (hunk 6 above)
Comment (gursheyss) [reply src/Pages/URLShortener/URLShortener.js:152]: done

**Review by pnavab — APPROVED** [review]
good

**Pull request merged**